react/week3/Todo.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from "react";
+import TodoDescription from "./TodoDescription";
+
+function Todo() {
+  const [todo, setTodo] = useState([{}]);
+  const ROOT_URL =
+    "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+  useEffect(() => {
+    (async () => {
+      const response = await fetch(ROOT_URL);
+      const result = await response.json();
+      setTodo(result);
+    })();
+  }, []);
+
+  const addTodo = (title, newDeadLine) => {
+    return setTodo([
+      ...todo,
+      {
+        id: todo[todo.length - 1]?.id + 1 || 1,
+        description: title,
+        deadline: newDeadLine,
+        isChecked: false,
+        isEdit: false
+      }
+    ]);
+  };
+  const toggleTodo = id => {
+    setTodo(prev =>
+      prev.map(todo => {
+        const newTodo = { ...todo };
+        if (todo.id === id) {
+          newTodo.isChecked = !newTodo.isChecked;
+        }
+        return newTodo;
+      })
+    );
+  };
+  const removeTodo = id => {
+    setTodo(prev =>

```

Comment:
We can change all the `prev` variables to `todos`. We should describe as better as possible what a variable means. `prev` is a very generic name.
```suggestion
    setTodo(todos =>

```